src/v/cluster/controller_backend.cc

```diff
@@ -810,6 +809,10 @@ ss::future<> controller_backend::reconcile_ntp_fiber(
                   ex);
             }
         }
+
+        // Only wait after we have made it through the semaphore once to avoid
+        // mass wakeups after the entering the function for the first time
+        co_await rs->wakeup_event.wait(_housekeeping_jitter.next_duration());
```

Comment:
So I think moving the wait to the end makes sense, but we have to modify the call sites of `reconcile_ntp_fiber` as well to avoid extra wakeups, i.e. instead of
```
event.set();
if (inserted) {
    spawn fiber
}
```
do
```
if (inserted) {
    spawn fiber
} else {
    event.set();
}
```